ISISComputingGroup/ibex_gui
# Ticket 3172: Added heartbeat check and some heartbeats

### Description of work

See https://github.com/ISISComputingGroup/IBEX/issues/3172

Contains changes from https://github.com/ISISComputingGroup/ibex_gui/pull/1057 so it can be run in CI.

### To Test

* Run the opi_checker script and confirm that the only heartbeat errors are on OPIs not written by ISIS

### Unit tests

Added tests in the opi_checker as well as in opi_checker_tests.

### System tests
N/A

### Documentation
N/A

---

#### Code Review


### Final Steps
